contracts/ProjectRegistry.sol

```diff
@@ -168,16 +168,23 @@ contract ProjectRegistry is Initializable, OwnableUpgradeable, ERC721Upgradeable
     }
 
     /**
-     * @notice update the deployment of a project, if in the restrict mode, only creator allowed call this function
+     * @notice update or add the deployment of a project, only creator allowed call this function
      */
-    function updateDeployment(uint256 projectId, bytes32 deploymentId, bytes32 metadata) external {
+    function addOrUpdateDeployment(uint256 projectId, bytes32 deploymentId, bytes32 metadata, bool updateLatest) external {
         require(ownerOf(projectId) == msg.sender, 'PR004');
-        require(deploymentInfos[deploymentId].projectId == 0, 'PR003');
-
-        projectInfos[projectId].latestDeploymentId = deploymentId;
-        deploymentInfos[deploymentId] = DeploymentInfo(projectId, metadata);
+        require(deploymentInfos[deploymentId].projectId == 0 || deploymentInfos[deploymentId].projectId == projectId, 'PR007');
+        bool changed = false;
+        if (deploymentInfos[deploymentId].metadata != metadata) {
+            deploymentInfos[deploymentId] = DeploymentInfo(projectId, metadata);
+            changed = true;
+        }
 
-        emit UpdateProjectDeployment(msg.sender, projectId, deploymentId, metadata);
+        if (updateLatest && projectInfos[projectId].latestDeploymentId != deploymentId) {
```

Comment:
Is it better to provide separate function to `updateProjectCurrentDeployment`. So that user can use it to update the provided deployment as the current project deployment only.

On the other hand `addOrUpdateDeployment` function only use to add and update deployment other than handle the case to `updateLatest`  only.

Reply:
How about I split this function into three
`function addDeployment(uint256 projectId, bytes32 deploymentId, bytes32 metadata, bool updateLatest)`
`function updateDeployment(uint256 projectId, bytes32 deploymentId, bytes32 metadata) external`
`function setProjectLatest(uint256 projectId, bytes32 deploymentId) external`
and a new event
`event UpdateProjectLatest(uint256 projectId, bytes32 deploymentId);`
@mzxyz

Reply:
Yeah, that would be better!